### PR TITLE
V3.0/file transfer

### PR DESCRIPTION
**Pull request by leefrank9527**
Added remote file transfering mechanism for distributed deployment and resolved some issues of gradle script:
1. Harvest Agent (H1 & H3) --> Store Component: 
  1) It's configurable to use "copy" mode or "stream" mode, and taking "copy" as the default mode.
  2) For "copy" mode, it's the existing method. When Heitrix and Store Component are deployed at the same machine, files can be copyed from none directory to another.
  3) For "stream" mode, it's the new added method. When Heitrix and Store Component are deployed at the different machines, files can be transfered via HTTP API.
  4) Got rid of the unused "save" APIs to simplify Store Component.
2. WebApp --> Store Component:
  When reading imported file from "uploadedFiles" folder at Store Component side, Store Component will check does the file exist. If the file does not exist, it will download the file from WebApp via HTTP API.
3. Update build.gradle:
  1) Resolved the deprecated warnings: to replace "classifier==''" with " archiveClassifier.set('')".
  2) Fixed file can't be found error for webapp/build.gradle of updateVersion function.

### CODE REVIEW

File: webcurator-db/latest/src/main/resources/setup/wct-create-mysql.sql [hunk 1]
```diff
@@ -2,5 +2,5 @@ CREATE DATABASE DB_WCT;
 \u DB_WCT
 create user usr_wct@localhost identified by 'usr_wct';
 grant all on DB_WCT.* to usr_wct@localhost;
-set password for usr_wct@localhost = PASSWORD('password');
-
+#set password for usr_wct@localhost = PASSWORD('password');
+alter user usr_wct@localhost identified by 'password';
```
Comment (obrienben) [inline webcurator-db/latest/src/main/resources/setup/wct-create-mysql.sql:6]: What issue was this fixing?

File: webcurator-webapp/build.gradle [hunk 2]
```diff
@@ -61,7 +59,7 @@ dependencies {
     providedRuntime 'org.springframework.boot:spring-boot-starter-tomcat'
 
     implementation 'com.h2database:h2:1.4.199'
-    implementation 'mysql:mysql-connector-java:5.1.6'
+    implementation 'mysql:mysql-connector-java:8.0.21'
```
Comment (obrienben) [inline webcurator-webapp/build.gradle]: What versions of mysql did you test this driver (8.0.21) with? Is it backwards compatible with older versions?

**Comment by leefrank9527** [comment]
@obrienben the two changes above about mysql should be ignored. I'm using Mysql 8 in my developemnt environment, and I didn't test that with Mysql 5.7.

**Comment by obrienben** [comment]
@leefrank9527 does this PR include the changes for H1 agent? There's no `digitalAssetStore.fileUploadMode` property in the H1 config files.

**Comment by leefrank9527** [comment]
H1 Agent should be included. The missing configuration was added, and was updated to the same branch: v3.0/file-transfer.

**Comment by obrienben** [comment]
Merged into master through the nlnz-uat-deploy branch as there were several merge conflicts and a couple of changes in this PR to revert.

**Pull request closed without merging**